packages/maps/src/index.ts

```diff
@@ -32,18 +53,31 @@ class Maps {
       version: '1.3.1'
     };
 
-    const requestOptions = {
+    return mapParameters;
+  }
+
+  private getRequestOptions(mapParameters: any) {
+    return {
       method: 'POST',
       headers: { 'Content-Type': 'application/json' },
       body: JSON.stringify(mapParameters)
     };
+  }
+
+  private fetchLayerGroup(requestOptions: any){
+    return fetch(`${this.apiBaseUrl()}?api_key=${this.apiKey}`, requestOptions)
+      .then((response) => response.json());
+  }
 
-    const { username, apiKey } = this._credentials;
+  private getVectorTileURL(layerGroup: any){
+    return `${this.apiBaseUrl()}/${layerGroup.layergroupid}/{z}/{x}/{y}.mvt?api_key=${this.apiKey}`;
+  }
 
-    return fetch(`https://${username}.carto.com/api/v1/map?api_key=${apiKey}`, requestOptions)
-      .then((response) => response.json())
-      .then((data) => `https://${username}.carto.com/api/v1/map/${data.layergroupid}/{z}/{x}/{y}.mvt?api_key=${apiKey}`);
+  private apiBaseUrl() {
```

Comment:
What about an organization user? I think that API's work without the user's organization URL, but just in case :)